src/main/resources/com/google/api/codegen/py/discovery_fragment.snip

```diff
@@ -8,51 +8,93 @@
+             'pip install --upgrade google-api-python-client'
+        @else
+          2. Install the Python client library for Google APIs by running
+             'pip install --upgrade google-api-python-client'
+        @end
         """
 
         @join import : imports
             {@import}
         @end
+        {@body}
+    @end
+@end
 
+@private adcBoilerplate()
         @# Authentication is provided by the 'gcloud' tool when running locally
         @# and by built-in service accounts when running on GAE, GCE, or GKE.
         @# See https://developers.google.com/identity/protocols/application-default-credentials for more information.
         credentials = GoogleCredentials.get_application_default()
-
-        @# Construct the {@apiName} service object (version {@apiVersion}) for interacting
-        @# with the API. You can browse other available API services and versions at
-        @# https://developers.google.com/api-client-library/python/apis/
-        {@serviceName()} = discovery.build('{@apiName}', '{@apiVersion}', credentials=credentials)
-        {@body}
-    @end
 @end
 
 @snippet generateBody(method)
     {@alwaysImport()}
-    @let params = context.getApiaryConfig.getMethodParams(method.getName)
-        {@initialize(method, params)}
-
-        @if context.hasRequestField(method)
-            {@initializeRequest(method)}
+    @let api = context.getApi, \
+         apiName = api.getName, \
+         apiVersion = api.getVersion, \
+         serviceTitle = context.getService.getTitle, \
+         apiaryConfig = context.getApiaryConfig, \
+         authType = apiaryConfig.getAuthType.toString
+      @switch authType
+      @case "APPLICATION_DEFAULT_CREDENTIALS"
+        {@importGoogleCredential()}{@adcBoilerplate()}
+
+      @case "OAUTH_3L"
+        @# {@TODO()} Implement this function to get authentication credentials.
+        @let authInstructionsUrl = apiaryConfig.getAuthInstructionsUrl
+          @if authInstructionsUrl
+            @# See {@authInstructionsUrl}
+          @end
         @end
-        @if context.isPageStreaming(method)
-            {@iterableCall(method)}
-        @else
-            {@compactCall(method)}
+        @if apiaryConfig.hasAuthScopes(method.getName)
+          @# Authorize using one of the following scopes in order to use this method:
+          @join scope : apiaryConfig.getAuthScopes.get(method.getName)
+            @#   {@scope}
+          @end
         @end
+        credentials = get_auth()
+
+        @default
+      @end
+
+      @# Construct the {@apiName} service object (version {@apiVersion}) for interacting
+      @# with the API. You can browse other available API services and versions at
+      @# https://developers.google.com/api-client-library/python/apis/
+      @if authType == "API_KEY"
+        @# {@TODO()} Replace api_key
+        {@serviceName()} = discovery.build('{@apiName}', '{@apiVersion}', developerKey="api_key")
```

Comment:
Is this the format we're using for placeholders? We keep switching it. Make sure it's consistent. Should it be `"{MY_API_KEY}"`? (and make the `TODO` above match)